src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,39 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.storage.Storage;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+import java.util.List;
+
+/**
+ * Represents a command entered by the user. Abstract base class to be inherited by
+ * more specific command types.
+ *
+ * @author Terence Chong Guang Jun
+ */
+public abstract class Command {
+    protected List<String> arguments;
+
+    Command() {
```

Comment:
this seems redundant

Reply:
Yes I wish it was also. I needed a package private constructor for the prototype instances of my concrete classes inheriting this. Could do without, but that would not be very good in terms of visibility.